icu4c/source/i18n/smpdtfmt.cpp

```diff
@@ -1324,46 +1292,15 @@ _appendSymbolWithMonthPattern(UnicodeString& dst, int32_t value, const UnicodeSt
 
 //----------------------------------------------------------------------
 
-static number::LocalizedNumberFormatter*
-createFastFormatter(const DecimalFormat* df, int32_t minInt, int32_t maxInt, UErrorCode& status) {
-    const number::LocalizedNumberFormatter* lnfBase = df->toNumberFormatter(status);
-    if (U_FAILURE(status)) {
-        return nullptr;
-    }
-    return lnfBase->integerWidth(
-        number::IntegerWidth::zeroFillTo(minInt).truncateAt(maxInt)
-    ).clone().orphan();
-}
-
-void SimpleDateFormat::initFastNumberFormatters(UErrorCode& status) {
-    if (U_FAILURE(status)) {
-        return;
-    }
-    auto* df = dynamic_cast<const DecimalFormat*>(fNumberFormat);
```

Comment:
Note: It looks like previously, the condition for making the fast formatters was if fNumberFormat is an instance of DecimalFormat. Try using this behavior instead of:

```
    // if it's null, we should not reinit since it might be using an adopted number formatter
```

Reply:
yep! planning to do this in response to markus' comment too.